### ISSUE TITLE
Maintenance: remove sqlalchemy mypy plugin

### ISSUE THREAD

**Issue by peterschutt** [starlite-api/starlite-pg-redis-docker#63]
Shouldn't need it with SQLAlchemy 2.0

**Comment by peterschutt** [comment]
> @peterschutt do you want "git+https://github.com/sqlalchemy/sqlalchemy" with types-SQLAlchemy or remove it at all ?

Sorry, not sure I follow. Are you asking if we should revert to sqlalchemy 1.4 and depend on the pypi version? I really wanted to test out the new api so built used this as an opportunity to do that.

I need to go through and update the way that the project is declaring the model columns, e.g., here:

```python
# app/core/model.py
@as_declarative(metadata=meta)
class Base:
    """Base for all SQLAlchemy declarative models."""

    ...

    id: Mapped["UUID"] = Column(pg.UUID, default=uuid4, primary_key=True)  # type:ignore[misc]
    created_date: Mapped["datetime"] = Column(DateTime, default=datetime.now, nullable=False)
    updated_date: Mapped["datetime"] = Column(DateTime, default=datetime.now, nullable=False)
```

should really be like this when using 2.0:

```python
class Base(DeclarativeBase):
    """
    Base for all SQLAlchemy declarative models.

    Attributes
    ----------
    created : Mapped[datetime]
        Date/time of instance creation.
    updated : Mapped[datetime]
        Date/time of last instance update.
    """

    registry = registry(
        metadata=MetaData(naming_convention=convention),
        type_annotation_map={UUID: pg.UUID, dict: pg.JSONB},
    )
    id: Mapped[UUID] = mapped_column(primary_key=True)
    created: Mapped[datetime] = mapped_column(default=datetime.now)
    updated: Mapped[datetime] = mapped_column(default=datetime.now)
```

Once that's done the stubs library and mypy plugin won't be necessary.

**Comment by grillazz** [comment]
Ok. Now I understand. Im happy to help with this refactor if you like

**Comment by peterschutt** [comment]
> Ok. Now I understand. Im happy to help with this refactor if you like

Sure, thanks!

Another thing that can change in this area is that `Mapped` now supports the `|` union operator - up until last week sometime it was required to use `Mapped[Optional[str]]` to denote a nullable column, but that can now be `Mapped[str | None]`.

**Comment by Goldziher** [comment]
try `pre-commit autoupdate`, it might be that an update to `pyupgrade` will simply update these on its own.

Also - make sure to update locally to python `3.10.7`.

**Comment by peterschutt** [comment]
closed in #73
